### ISSUE TITLE
Add "latest version" button

### ISSUE THREAD

**Issue by tmorrell** [inveniosoftware/invenio-app-rdm#2655]
## Is your feature request related to a problem? Please describe.
Currently the latest version is determined by the last record that was added to the system. However, when importing records or updating historic records it is useful to select a specific record version as the latest.

## Describe the solution you'd like
The addition of a new button "Set as latest version" on the record edit screen when multiple versions are present. This would set 
 `versions.is_latest: true`, and have the selected version show up in the search results. There should be a configuration option for this functionality

## Describe alternatives you've considered
This could also be an API-only functionality.

**Comment by ntarocco** [comment]
I believe this should be an operation only allowed to admins. Or, do you think that the record owner should be able to do it?

There could also be a temporal problem, where the latest version could have been created before the second latest. Would this be confusing to the user?

**Comment by tmorrell** [comment]
Yup, I think this should be an admin or special permission functionality.

The current behavior works great for a self-service data repository, but for a general IR you might end up adding older published content after newer published content. In this case the "created in RDM" date isn't the date you want to use for "latest". Instead of trying to code in some complicated date handling system, a button would allow admins to manually fix issues.

**Comment by tmorrell** [comment]
This functionality would also need to adjust the index of the versions to correct the order https://github.com/inveniosoftware/invenio-app-rdm/issues/2955. We'd ideally like admins to be able to reorder versions, but there is some complexity in getting the reordering correct.
